src/workerd/util/co-capture.h

```diff
@@ -4,48 +4,48 @@
 
 #pragma once
 
-#include <kj/debug.h>
-#include <type_traits>
+#include <kj/common.h>
 
 namespace workerd {
 
 namespace _ {
+
 template<typename Functor>
 struct CaptureForCoroutine {
   kj::Maybe<Functor> maybeFunctor;
 
   explicit CaptureForCoroutine(Functor&& f) : maybeFunctor(kj::mv(f)) {}
 
   template<typename ...Args>
-  using ReturnType = std::invoke_result_t<Functor, Args...>;
-
-  template<typename ...Args>
-  static auto coInvoke(Functor functor, Args&&... args) -> ReturnType<Args&&...> {
+  static auto coInvoke(Functor functor, Args&&... args)
```

Comment:
the name functor here confuses me a lot.

As far as I remember the theory, in expression `map(list, f)` - `map` is a functor. `f` is a function.
Or more exactly: `map(*, f)` is a list->list functor. (https://en.wikipedia.org/wiki/Functor_(functional_programming)#:~:text=In%20functional%20programming%2C%20a%20functor,in%20Haskell%20using%20type%20class)

So in this context CaptureForCoroutine is a functor. And it operates on functions.

Wdyt?

Reply:
I agree with you, but the misnomer is already quite prevalent in this codebase. I refrained from changing the names (I prefer "function object" myself) to avoid opening the can of worms. :)

Reply:
It is a misnomer, but it's a broader C++ community misnomer, not one unique to this code. That wiki page itself hints at why it ended up getting used here:

> In [C++](https://en.wikipedia.org/wiki/C%2B%2B), the name functor is commonly used to refer to a [function object](https://en.wikipedia.org/wiki/Function_object), even though the ISO/IEC 14882 standard specification itself exclusively uses the latter term.

I don't think anyone would be too upset if you just renamed this to `fn` though.